### ISSUE TITLE
Last few GCS when md files is to much

### ISSUE THREAD

**Issue by xinconan** [vuejs/vitepress#2214]
### Describe the bug

As the number of files increases, the memory usage increases. The case I has nearly 1000 md files, build the docs then cause few GCs error. I used `node --max_old_space_size=6144 node_modules/vitepress/bin/vitepress.js build docs`, I can't add the size to more.
```
<--- Last few GCs --->

[33180:0000020AC2411ED0]   306683 ms: Scavenge (reduce) 6033.4 (6190.7) -> 6033.1 (6191.2) MB, 9.9 / 0.0 ms  (average mu = 0.154, current 
mu = 0.001) allocation failure
[33180:0000020AC2411ED0]   313287 ms: Mark-sweep (reduce) 6034.2 (6191.6) -> 6033.0 (6192.1) MB, 6596.8 / 0.0 ms  (average mu = 0.085, current mu = 0.008) allocation failure scavenge might not succeed
```

### Reproduction

create 1000 md files, then reproduction.

### Expected behavior

It can build normally when use 6G memory

### System Info

```shell
window 10 64, vitepress@1.0.0-alpha.65
```

### Validations

- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.dev).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

**Comment by MarkusKeck** [comment]
Is the build itself successfull or does the whole build fail?

**Comment by xinconan** [comment]
> Is the build itself successfull or does the whole build fail?

whole build fail

**Comment by brc-dd** [comment]
fixed in #3366
